src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,44 +506,323 @@ public int getID()
     }
 
     /**
-     * Increment the mobs killed by this colony.
-     * <p>
-     * Will award achievements for mobs killed.
+     * Get the amount of statistic.
+     *
+     * @param statistic the statistic.
+     * @return amount of statistic.
      */
-    public void incrementMobsKilled()
+    private int getStatisticAmount(@NotNull String statistic)
     {
-        killedMobs++;
-        final int mobKills = this.getKilledMobs();
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIRST)
+        switch (statistic)
         {
-            this.triggerAchievement(ModAchievements.achievementKillOneMob);
+            case TAG_GUARD_MOBS:
+                return killedMobs;
+            case TAG_MINER_ORES:
+                return minedOres;
+            case TAG_MINER_DIAMONDS:
+                return minedDiamonds;
+            case TAG_BUILDER_HUTS:
+                return builtHuts;
+            case TAG_FISHERMAN_FISH:
+                return caughtFish;
+            case TAG_FARMER_WHEAT:
+                return harvestedWheat;
+            case TAG_FARMER_POTATOES:
+                return harvestedPotatoes;
+            case TAG_FARMER_CARROTS:
+                return harvestedCarrots;
+            case TAG_LUMBERJACK_SAPLINGS:
+                return plantedSaplings;
+            case TAG_LUMBERJACK_TREES:
+                return felledTrees;
+            default:
+                return 0;
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_SECOND)
+    }
+
+    /**
+     * Trigger first achievement.
+     * @param statistic the statistic.
+     */
+    private void triggerFirstAchievement(@NotNull String statistic)
```

Comment:
can we move all these to an utility class? they bloat up the already big Colony class